www/js/diary/list.js

```diff
@@ -172,16 +172,26 @@ angular.module('emission.main.diary.list',['ui-leaflet',
 
     $scope.$on(Timeline.UPDATE_DONE, function(event, args) {
       console.log("Got timeline update done event with args "+JSON.stringify(args));
-      $scope.$apply(function() {
-          $scope.data = Timeline.data;
-          $scope.datepickerObject.inputDate = Timeline.data.currDay.toDate();
-          $scope.data.currDayTrips.forEach(function(trip, index, array) {
-              PostTripManualMarker.addUnpushedIncidents(trip);
-          });
-          $scope.data.currDayTripWrappers = Timeline.data.currDayTrips.map(
-            DiaryHelper.directiveForTrip);
-          $ionicScrollDelegate.scrollTop(true);
-      });
+      if ($rootScope.tripTimelineUpdate == true) {
```

Comment:
When I tested it, I added a new `$scope.$on(Timeline.UPDATE_DONE, function(event, args) {` to `metrics.js` and put this new code (that invokes `Timeline.updateForDay($rootScope.recentTripDate.startOf('day'));...`) in there.

Reply:
You should do something like that too because otherwise, you call the update when it is done, which seems wrong.